### DISCUSSION TITLE
Using external libraries (vue-wamp) with Quasar

### DISCUSSION THREAD

**Discussion by FadhiliNjagi** [quasarframework/quasar#11756] [General - Components / Directives / etc] [answered]
I'm kind of new to Quasar, and I want to use the [lajosbencz/vue-wamp](https://github.com/lajosbencz/vue-wamp) module, but I don't know how. I have been looking at Quasar boot files but I don't know exactly what to do, and the documentation is a bit vague.

In a normal Vue main.js file, I could paste the following code:

```
import VueWamp from 'vue-wamp'

Vue.use(VueWamp, {
    url: 'ws://demo.crossbar.io/ws',
    realm: 'realm1',

    // change this in case of naming conflict
    namespace: 'wamp',
    // automatically re-registers and re-subscribes after reconnection (on by default)
    auto_reestablish: true,
    // automatically closes WS connection after amount of idle milliseconds (off by default)
    auto_close_timeout: -1,
});
```

Please advise me on how to do this the Quasar way. I can't seem to find an answer anywhere on the web.
**PS:** I'm using TypeScript.

**Comment by pdanpdan** [comment, marked as answer]
From the syntax it looks like it's a Vue2 component (Quasar v2 is using Vue3) - check if you have an updated component for Vue3

**Reply by FadhiliNjagi** [reply]
Yeah, you're right. The last release of that module was in 2019, way before Vue 3 was released. I'll look for another solution, thanks.
